docker-entrypoint.sh

```diff
@@ -588,13 +591,17 @@ _main() {
 			sleep 5
 		done
 	else
-		if [ "x${PGAUTO_ONESHOT}" = "xyes" ]; then
-			echo "*****************************************************************************************************"
-			echo "'One shot' automatic upgrade was requested, so exiting now rather than starting the PostgreSQL server"
-			echo "*****************************************************************************************************"
-		else
-			exec "$@"
+		# If the upgrade process ran, then we need to launch the post-upgrade script in the background
+		if [ "${UPGRADE_PERFORMED}" -eq 1 ]; then
+			TIMESTAMP_NOW=$(date +'%Y.%m.%d-%H.%M')
+			echo "**********************************************************************************************************************"
+			/usr/local/bin/pgautoupgrade-postupgrade.sh "${PGDATA}" "${PGAUTO_ONESHOT}" 2>&1 | tee "${PGDATA}/${TIMESTAMP_NOW}-pgautoupgrade.log" &
+			echo "Post upgrade script launched, with output being saved to ${PGDATA}/${TIMESTAMP_NOW}-pgautoupgrade.log in the container"
+			echo "**********************************************************************************************************************"
 		fi
+
+		# Start PostgreSQL
+		exec "$@"
```

Comment:
if you start the container with `PGAUTO_ONESHOT` set to `yes`, but no upgrade is required, then the container will just start as usual. I think previously, in any case, the container shut down.

Reply:
Oh, good point.  I'll rework the logic to fix that.  Probably won't be tonight (need sleep) but it shouldn't be difficult. :smile:

Reply:
Fixed this logic bug.  Went with a simple approach that provides useful output to the container user, as that seems like it'll minimise problems for our users.